common/persistence/nosql/nosql_history_store_test.go

```diff
@@ -527,47 +527,630 @@ func TestDeleteHistoryBranch_unusedBranch(t *testing.T) {
+// | 								 │     Tree: 123     │
+// | 								 │     Branch: B     │
+// | 								 │     Node: 3       │
+// | 								 └─────────┬─────────┘
+// | 								           │
+// | 								           │
+// | 								 ┌─────────▼─────────┐
+// | 								 │     Tree: 123     │
+// | 								 │     Branch: B     │
+// | 								 │     Node: 4       │
+// | 								 └───────────────────┘
+// |
+// |  The expected behaviour, is that the child/branched workflow needs to clean up both its own history nodes
+// |  but *all* of the parent's remaining and now unreferenced history nodes. They're otherwise unreachable
+// |  and will be just history_node table garbage which forever grows.
+// |
+// | 	Trees
+// |
+// | 	┌─ ─ ─ ─ ─ ─ ─ ─ ─ ─┐    ┌─ ─ ─ ─ ─ ─ ─ ─ ─ ─┐
+// | 	                         │                   │
+// | 	│  <branch A is     │
+// | 	    deleted>             │   deleted>        │
+// | 	│                   │
+// | 	                         │                   │
+// | 	└─ ─ ─ ─ ─ ─ ─ ─ ─ ─┘    └─ ─ ─ ─ ─ ─ ─ ─ ─ ─┘
+// |
+// |
+// | 	Nodes
+// |
+// | 	┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
+// | 	│                   │
+// | 	│  <deleted>        │
+// | 	│                   │
+// | 	└ ─ ─ ─ ─ ┌ ─ ─ ─ ─ ┘
+// | 	          │
+// | 	          ┼ ─ ─ ─ ─ ── ─ ── ─ ─ ── ┐
+// | 	┌ ─ ─ ─ ─ ▼ ─ ─ ─ ─ ┐    ┌─ ─ ─ ── ┼─ ─ ─ ─ ─┐
+// | 	│                   │    │                   │
+// | 	│  <deleted>        │    │  <deleted>        │
+// | 	│                   │    │                   │
+// | 	└ ─ ─ ─ ─ ┌ ─ ─ ─ ─ ┘    └─ ─ ─ ── ┌─ ─ ─ ─ ─┘
+// | 	          │                        │
+// | 	          │                        │
+// | 	┌ ─ ─ ─ ─ ▼ ─ ─ ─ ─ ┐    ┌─ ─ ─ ── ▼─ ─ ─ ─ ─┐
+// | 	│                   │    │                   │
+// | 	│  <deleted>        │    │ <deleted>         │
+// | 	│                   │    │                   │
+// | 	└ ─ ─ ─ ─ ┌ ─ ─ ─ ─ ┘    └─ ─ ─ ── ┌─ ─ ─ ─ ─┘
+// | 	          │                        │
+// | 	          │                        │
+// | 	┌ ─ ─ ─ ─ ▼ ─ ─ ─ ─ ┐    ┌─ ─ ─ ── ▼─ ─ ─ ─ ─┐
+// | 	│                   │    │                   │
+// | 	│  <deleted>        │    │ <deleted>         │
+// | 	│                   │    │                   │
+// | 	└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘    └─ ─ ─ ── ── ─ ─ ─ ─┘
+func TestDeleteHistoryBranch_usedBranch(t *testing.T) {
+	t.Skip("not implemented yet")
+	store, dbMock, _ := setUpMocks(t)
+
+	request := &persistence.InternalDeleteHistoryBranchRequest{
+		BranchInfo: types.HistoryBranch{
+			TreeID:   "TestTreeID",
+			BranchID: "B",
+			Ancestors: []*types.HistoryBranchRange{
+				{
+					BranchID:  "A",
+					EndNodeID: 3,
+				},
+			},
+		},
+		ShardID: testShardID,
+	}
+
+	expectedTreeFilter := &nosqlplugin.HistoryTreeFilter{
+		ShardID:  testShardID,
+		TreeID:   "TestTreeID",
+		BranchID: common.Ptr("B"),
+	}
+
+	expectedNodeFilters := []*nosqlplugin.HistoryNodeFilter{
+		{
+			ShardID:   testShardID,
+			TreeID:    "TestTreeID",
+			BranchID:  "A",
+			MinNodeID: 3,
+		},
```

Comment:
```suggestion
		{
			ShardID:   testShardID,
			TreeID:    "TestTreeID",
			BranchID:  "A",
		},
```
I believe the final version will be like this, no min-node?  (not at all demanding precision for a not-yet-working test, kinda just sanity checking)